master/internal/resourcemanagers/agent/agent.go

```diff
@@ -343,3 +371,68 @@ func (a *agent) summarize(ctx *actor.Context) model.AgentSummary {
 		Draining:       a.draining,
 	}
 }
+
+func (a *agent) gatherContainersToReattach(ctx *actor.Context) []aproto.ContainerReattach {
+	result := make([]aproto.ContainerReattach, 0, len(a.containers))
+	for containerID, allocation := range a.containers {
+		resp := ctx.Ask(allocation, sproto.GetTaskContainerState{
```

Comment:
needing this ask is a bit much, side effect of the master architecture though (which, of course, i'm determined to fix). asks like this become a bit dangerous, we might should start a list somewhere, so we can statically check for deadlock potential